File: experiments_records/1_latent vulnerabilities/1_project_level/js/educollab_js_mvc/src/repos/submissionRepo.js
const { db, nowIso } = require('../db');

function createSubmission({ assignment_id, course_id, student_id, content_text, attachment_upload_id }) {
  const ts = nowIso();
  const info = db.prepare(
    'INSERT INTO submissions (assignment_id, course_id, student_id, content_text, attachment_upload_id, created_at, updated_at) VALUES (?,?,?,?,?,?,?)'
  ).run(assignment_id, course_id, student_id, content_text, attachment_upload_id || null, ts, ts);
  return getById(course_id, assignment_id, info.lastInsertRowid);
}

function getById(course_id, assignment_id, submission_id) {
  return db.prepare(`
    SELECT s.*, u.username as student_username
    FROM submissions s
    JOIN users u ON u.user_id = s.student_id
    WHERE s.course_id=? AND s.assignment_id=? AND s.submission_id=?
  `).get(course_id, assignment_id, submission_id);
}

function getMySubmission(course_id, assignment_id, student_id) {
  return db.prepare(`
    SELECT s.*
    FROM submissions s
    WHERE s.course_id=? AND s.assignment_id=? AND s.student_id=?
  `).get(course_id, assignment_id, student_id);
}

function updateSubmission(course_id, assignment_id, submission_id, { content_text, attachment_upload_id }) {
  db.prepare(
    'UPDATE submissions SET content_text=?, attachment_upload_id=?, updated_at=? WHERE course_id=? AND assignment_id=? AND submission_id=?'
  ).run(content_text, attachment_upload_id || null, nowIso(), course_id, assignment_id, submission_id);
  return getById(course_id, assignment_id, submission_id);
}

function listMySubmissions(student_id) {
  return db.prepare(`
    SELECT s.*, a.title as assignment_title, c.title as course_title
    FROM submissions s
    JOIN assignments a ON a.assignment_id = s.assignment_id
    JOIN courses c ON c.course_id = s.course_id
    WHERE s.student_id=?
    ORDER BY s.updated_at DESC
  `).all(student_id);
}

function listForAssignment(course_id, assignment_id) {
  return db.prepare(`
    SELECT s.*, u.username as student_username
    FROM submissions s
    JOIN users u ON u.user_id = s.student_id
    WHERE s.course_id=? AND s.assignment_id=?
    ORDER BY s.updated_at DESC
  `).all(course_id, assignment_id);
}

module.exports = {
  createSubmission,
  getById,
  getMySubmission,
  updateSubmission,
  listMySubmissions,
  listForAssignment
};

File: experiments_records/1_latent vulnerabilities/2_requirement_level/python/educollab_flask_req04_repair_iteration4/models/submission_repo.py
from models.db import get_db



def create_submission(assignment_id, course_id, student_id, content_text, attachment_upload_id=None):
    db = get_db()
    cursor = db.execute(
        'INSERT INTO submissions (assignment_id, course_id, student_id, content_text, attachment_upload_id) VALUES (?, ?, ?, ?, ?)',
        (assignment_id, course_id, student_id, content_text, attachment_upload_id),
    )
    db.commit()
    return get_submission_by_id(course_id, assignment_id, cursor.lastrowid)


def get_submission_by_id(course_id, assignment_id, submission_id):
    db = get_db()
    return db.execute(
        'SELECT s.submission_id, s.assignment_id, s.course_id, s.student_id, s.content_text, s.attachment_upload_id, s.created_at, s.updated_at, '
        'u.username, u.display_name, up.original_name AS attachment_name '
        'FROM submissions s '
        'JOIN users u ON u.user_id = s.student_id '
        'LEFT JOIN uploads up ON up.upload_id = s.attachment_upload_id '
        'WHERE s.course_id = ? AND s.assignment_id = ? AND s.submission_id = ?',
        (course_id, assignment_id, submission_id),
    ).fetchone()


def update_submission(course_id, assignment_id, submission_id, content_text, attachment_upload_id=None):
    db = get_db()
    db.execute(
        'UPDATE submissions SET content_text = ?, attachment_upload_id = ?, updated_at = CURRENT_TIMESTAMP WHERE course_id = ? AND assignment_id = ? AND submission_id = ?',
        (content_text, attachment_upload_id, course_id, assignment_id, submission_id),
    )
    db.commit()
    return get_submission_by_id(course_id, assignment_id, submission_id)


def list_submissions_for_assignment(course_id, assignment_id):
    db = get_db()
    return db.execute(
        'SELECT s.submission_id, s.assignment_id, s.course_id, s.student_id, s.content_text, s.attachment_upload_id, s.created_at, s.updated_at, '
        'u.username, u.display_name, up.original_name AS attachment_name '
        'FROM submissions s '
        'JOIN users u ON u.user_id = s.student_id '
        'LEFT JOIN uploads up ON up.upload_id = s.attachment_upload_id '
        'WHERE s.course_id = ? AND s.assignment_id = ? ORDER BY s.submission_id DESC',
        (course_id, assignment_id),
    ).fetchall()


def list_submissions_for_student(student_id):
    db = get_db()
    return db.execute(
        'SELECT s.submission_id, s.assignment_id, s.course_id, s.student_id, s.content_text, s.attachment_upload_id, s.created_at, s.updated_at, '
        'u.username, u.display_name, a.title AS assignment_title, c.title AS course_title, up.original_name AS attachment_name '
        'FROM submissions s '
        'JOIN users u ON u.user_id = s.student_id '
        'JOIN assignments a ON a.assignment_id = s.assignment_id '
        'JOIN courses c ON c.course_id = s.course_id '
        'LEFT JOIN uploads up ON up.upload_id = s.attachment_upload_id '
        'WHERE s.student_id = ? ORDER BY s.submission_id DESC',
        (student_id,),
    ).fetchall()